src/interactive/repl.ts

```diff
@@ -25,6 +26,25 @@ let g_terminal: vscode.Terminal = null
 
 export let g_connection: rpc.MessageConnection = undefined
 
+function isConnectionActive() {
+    return g_connection !== undefined
+}
+
+export async function withREPL(
+    callback: (connection: rpc.MessageConnection) => any,
+    callbackOnHandledErr: (err: Error) => any
+) {
+    if (!isConnectionActive()) {
+        return callbackOnHandledErr(new Error('REPL is not active'))
+    }
+    try {
+        return callback(g_connection)
+    } catch (err) {
+        // TODO: identify which error should be handled
+        throw err
+    }
+}
```

Comment:
Same here, I'm not a huge fan of these callback style APIs. But either way, if we do introduce that into the codebase, we should do it in a separate PR.

Reply:
Actually `withREPL` is now not used anywhere in this PR, so I will remove this one.
Well, as for the callback API (i..e. `withLanguageClient` in this PR), I would like to use such a thing when we don't need to subscribe to frequent events, which may need frequent access to the connection.
The main benefit is that we don't need any global (`g_languageClient` and such), and we can focus on error handling at one place.